RocketChat/Rocket.Chat.ReactNative
# regression: Status bar's color theme is incorrect

## Proposed changes


The PR https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/5132 fixed the issue https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/4889, however, it introduced a bug when the user opens the AttachmentView and the status bar kept the color of AttachmentView Status bar until re-render the status bar of a new view. So, this PR is intended to fix this regression and still fix the issue related. Now the approach was to remove the `React.memo`, and let the component render together with his father.

## Issue(s)	


## How to test or reproduce


- Open the AttachmentView with Android, the status bar color should be expected for each theme
- Test the other [PR's](https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/4889) test case:

This error it's happening on the first open at my Samsung Galaxy A31. It's working properly with Xiaomi Redmi 9 and with iOS devices.

>  - OS in light mode.
>  - Open RocketChat => status bar text is light (but should be dark)
>  - Change OS to dark mode via Quick Settings Tile
>  - RocketChat status bar text is now light ✅
>  - Change OS to light mode
>  - RocketChat status bar text is now dark ✅
>  - Close RocketChat and remove it from Recent Apps
>  - Open RocketChat again
>  - RocketChat status bar text is now light again (but should be dark)

## Screenshots

#### Before
https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/3736147d-43fc-441a-85cd-915510d2320b

#### After
https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/641f6360-287a-4511-bf3f-230a22fb0b46


#### Test if the fix before still working
https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/6a8330a3-6718-4245-8f35-13c2541eb20f


## Types of changes




## Checklist



## Further comments

[TC-886]

[TC-886]: https://rocketchat.atlassian.net/browse/TC-886?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
